File: java/src/main/java/compass/main/ExampleClient.java
package compass.main;

import com.google.protobuf.Timestamp;
import compass.harvest.v1alpha1.IngestServiceGrpc;
import compass.harvest.v1alpha1.PositionEventOuterClass.*;
import compass.type.geo.Geo;
import io.grpc.*;
import io.grpc.stub.MetadataUtils;

public class ExampleClient {

    public static void main(String[] args) {

        String url = "harvest.grpc.apis.compassiot.cloud";
        String apiKey = "apikey bc8760b6f6f12b8ada9fdfd4503e847f";

        Channel ch = ManagedChannelBuilder.forAddress(url, 443).build();
        IngestServiceGrpc.IngestServiceBlockingStub client = IngestServiceGrpc.newBlockingStub(ch);

        Metadata header = new Metadata();
        Metadata.Key<String> key = Metadata.Key.of("authorization", Metadata.ASCII_STRING_MARSHALLER);
        header.put(key, apiKey);
        client = MetadataUtils.attachHeaders(client, header);

        PositionEvent evt = PositionEvent
                .newBuilder()
                .setVehicleId("abc123")
                .setVehicleType(VehicleType.CAR)
                .setTransportType(TransportType.PRIVATE)
                .setTimestamp(Timestamp
                        .newBuilder()
                        .setSeconds(System.currentTimeMillis() / 1000)
                        .build())
                .setAcceleration(Acceleration
                        .newBuilder()
                        .setX(1)
                        .setY(2)
                        .setZ(3)
                        .build())
                .setPosition(Position
                        .newBuilder()
                        .setLatlng(Geo.LatLng
                                .newBuilder()
                                .setLatitude(45)
                                .setLongitude(90))
                        .setSpeed(60)
                )
                .setGyro(Gyro
                        .newBuilder()
                        .setRoll(-1)
                        .setPitch(-2)
                        .setYaw(-3))
                .putLabels("make", "toyota")
                .putLabels("something", "nice")
                .build();
        try {
            client.insert(evt);
        } catch (Exception e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
        System.out.println("Success!");
    }
}

File: streaming/csharp/fixmeharrison/ExampleClient.cs
using Google;
using Grpc;
using Compass.Harvest.V1Alpha1;
using Compass.Type.Geo;
using Compass.Iam;
using Grpc.Net.Client;
using Google.Protobuf.WellKnownTypes;


namespace Client
{
    class ExampleClient
    {
        static void Main(string[] args)
        {
            
        String url = "https://harvest.grpc.apis.compassiot.cloud:443";
        String apiKey = "apikey bc8760b6f6f12b8ada9fdfd4503e847f";
        GrpcChannel channel = GrpcChannel.ForAddress(url);
        IngestService.IngestServiceClient client = new Compass.Harvest.V1Alpha1.IngestService.IngestServiceClient(channel);
        Grpc.Core.Metadata headers = new Grpc.Core.Metadata();
        headers.Add("Authorization", $"{apiKey}");
        Grpc.Core.Metadata header = new Grpc.Core.Metadata();
        
        PositionEvent evt = new PositionEvent { 
            VehicleId = "testvehicle123", 
            VehicleType = VehicleType.Car, 
            TransportType = TransportType.Private, 
            Acceleration =  new Acceleration {X = 1, Y = 2, Z = 3}, 
            Position = new Position{Latlng = new LatLng{Latitude = 45, Longitude = 90}},
            Gyro = new Gyro{Roll = -1, Pitch = -2, Yaw = -3},
            Timestamp = Timestamp.FromDateTime(DateTime.UtcNow)};
            

        
        try
        {
            client.Insert(evt, headers);
            client.Insert(evt2, headers);
        }
        catch (Exception e)
        {
            Console.Write(e.Message);
            return 0;
        }
            Console.Write("\nSuccessfully pushed event via GRPC API!");
            return 0;
        }
    }
}